perses/annihilation/ncmc_switching.py

```diff
@@ -734,212 +767,121 @@ def integrate(self, topology_proposal, initial_positions, proposed_positions, pl
 
-class NCMCAlchemicalIntegrator(openmm.CustomIntegrator):
-    """
-    Helper base class for NCMC alchemical integrators.
+
+class NCMCAlchemicalIntegrator(NonequilibriumLangevinIntegrator):
+    """Allows nonequilibrium switching based on force parameters specified in alchemical_functions.
+    A variable named lambda is switched from 0 to 1 linearly throughout the nsteps of the protocol.
+    The functions can use this to create more complex protocols for other global parameters.
+    Propagator is based on Langevin splitting, as described below.
+    One way to divide the Langevin system is into three parts which can each be solved "exactly:"
+        - R: Linear "drift" / Constrained "drift"
+            Deterministic update of *positions*, using current velocities
+            x <- x + v dt
+        - V: Linear "kick" / Constrained "kick"
+            Deterministic update of *velocities*, using current forces
+            v <- v + (f/m) dt
+                where f = force, m = mass
+        - O: Ornstein-Uhlenbeck
+            Stochastic update of velocities, simulating interaction with a heat bath
+            v <- av + b sqrt(kT/m) R
+                where
+                a = e^(-gamma dt)
+                b = sqrt(1 - e^(-2gamma dt))
+                R is i.i.d. standard normal
+    We can then construct integrators by solving each part for a certain timestep in sequence.
+    (We can further split up the V step by force group, evaluating cheap but fast-fluctuating
+    forces more frequently than expensive but slow-fluctuating forces. Since forces are only
+    evaluated in the V step, we represent this by including in our "alphabet" V0, V1, ...)
+    When the system contains holonomic constraints, these steps are confined to the constraint
+    manifold.
+    Examples
+    --------
+        - VVVR
+            splitting="O V R V O"
+        - BAOAB:
+            splitting="V R O R V"
+        - g-BAOAB, with K_r=3:
+            splitting="V R R R O R R R V"
+        - g-BAOAB with solvent-solute splitting, K_r=K_p=2:
+            splitting="V0 V1 R R O R R V1 R R O R R V1 V0"
+        - An NCMC algorithm with Metropolized integrator:
+            splitting="O { V R H R V } O"
+    Attributes
+    ----------
+    _kinetic_energy : str
+        This is 0.5*m*v*v by default, and is the expression used for the kinetic energy
+    References
+    ----------
+    [Nilmeier, et al. 2011] Nonequilibrium candidate Monte Carlo is an efficient tool for equilibrium simulation
+    [Leimkuhler and Matthews, 2015] Molecular dynamics: with deterministic and stochastic numerical methods, Chapter 7
     """
-    def __init__(self, temperature, system, functions, nsteps, steps_per_propagation, timestep, direction):
-        """
-        Initialize base class for NCMC alchemical integrators.
 
+    def __init__(self,
+                 alchemical_functions,
+                 splitting="O { V R H R V } O",
```

Comment:
Shouldn't you ensure that the `splitting` contains at least one `H` ? Otherwise, I think `NCMCAlchemicalIntegrator` won't do any alchemy.

Also, don't we want to default to g-BAOAB?